action/protocol/staking/handlers.go

```diff
@@ -551,18 +583,30 @@ func (p *Protocol) fetchBucket(
 	index uint64,
 	checkOwner bool,
 	allowSelfStaking bool,
-) (*VoteBucket, error) {
+) (*VoteBucket, *fetchError) {
 	actionCtx := protocol.MustGetActionCtx(ctx)
 	bucket, err := getBucket(sr, index)
 	if err != nil {
```

Comment:
i think need to further differentiate the error

if err.Cause == state.ErrNotExist, then we should return ReceiptStatus_ErrInvalidBucketIndex, and settle the action as Failed, that is charge the gas, and Mint/CommitBlock() can proceed

otherwise, the error would be a real error (like DB failed), in that case, Mint/CommitBlock should fail/stop

and at the return of fetchBucket(), need to tell apart these 2 cases, and process accordingly

Reply:
agreed.